hail/python/hail/matrixtable.py

```diff
@@ -224,16 +227,133 @@ def partition_hint(self, n: int) -> 'GroupedMatrixTable':
         return self
 
     @typecheck_method(named_exprs=expr_any)
-    def aggregate(self, **named_exprs) -> 'MatrixTable':
-        """Aggregate by group, used after :meth:`.MatrixTable.group_rows_by` or :meth:`.MatrixTable.group_cols_by`.
+    def aggregate_cols(self, **named_exprs) -> 'GroupedMatrixTable':
+        """Aggregate cols by group.
+
+        Examples
+        --------
+        Aggregate to a matrix with cohort as column keys, computing the mean height
+        per cohort as a new column field:
+
+        >>> dataset_result = (dataset.group_cols_by(dataset.cohort)
+        ...                          .aggregate_cols(mean_height = agg.stats(dataset.pheno.height).mean)
+        ...                          .result())
```

Comment:
I kinda think this should be an error, or should return a Table.  Thoughts?

Reply:
I think it would be confusing if it returns a table in some cases and a matrix table in others. An error might be ok, but I think the current interface is fine. Maybe a warning that no entry fields are defined?